pkg/chunk/local/boltdb_table_client.go

```diff
@@ -2,27 +2,45 @@ package local
 
 import (
 	"context"
+	"os"
+	"path/filepath"
 
 	"github.com/cortexproject/cortex/pkg/chunk"
 )
 
-type tableClient struct{}
+type tableClient struct {
+	directory string
+}
```

Comment:
Question: would it make sense to have a single implementation (ie singleton) that implements both TableClient, and StorageClient?  That way DeleteTable can close the active table instance properly.

Reply:
Hrm, while that would work if the same client is being used for queries and table-manager, but I don't think that is the case. I think two different clients are instantiated for store and table-manager.